### PR TITLE
Rework logstorage

### PR DESCRIPTION
**Pull request by minminlittleshrimp**
+ Rework logstorage, activate all unit test
+ Allow rerun failed test cases in verbose mode in Github Action
+ Adapt dlt-logd-converter test environment setup

See https://github.com/COVESA/dlt-daemon/pull/534

### CODE REVIEW

**Comment by minminlittleshrimp** [comment]
Please kindly have a look in this verbose logs of the test result: 
[ctest_full_log.txt](https://github.com/COVESA/dlt-daemon/files/12796383/ctest_full_log.txt)

**Review by lvklevankhanh — APPROVED** [review]

**Review by lvklevankhanh — APPROVED** [review]

File: src/offlinelogstorage/dlt_offline_logstorage_behavior.c [hunk 1]
```diff
@@ -50,7 +50,7 @@ DLT_STATIC void dlt_logstorage_concat_logfile_name(char *log_file_name, const ch
     size_t src_len = strlen(append);
 
     if (dst_len < DLT_MOUNT_PATH_MAX) {
-        size_t rem_len = DLT_MOUNT_PATH_MAX - dst_len - 1;
+        size_t rem_len = DLT_MOUNT_PATH_MAX - dst_len;
```
Comment (michael-methner) [inline src/offlinelogstorage/dlt_offline_logstorage_behavior.c:53]: Please revert this change. 
strncat() below expects:  "destination buffer large enough, that is, strlen(dst) + strnlen(src, sz) + 1." [1] 


[1] https://man7.org/linux/man-pages/man3/strncat.3.html

File: src/offlinelogstorage/dlt_offline_logstorage_behavior.c [hunk 2]
```diff
@@ -371,20 +370,15 @@ int dlt_logstorage_storage_dir_info(DltLogStorageUserConfig *file_config,
...
 
-    char suffix[10];
-    int suffix_len;
-    memset(suffix, 0, 10);
-    if (config->gzip_compression) {
-        suffix_len = DLT_OFFLINE_LOGSTORAGE_GZ_FILE_EXTENSION_LEN;
-        strncpy(suffix, ".dlt.gz", suffix_len);
-    }
-    else {
-        suffix_len = DLT_OFFLINE_LOGSTORAGE_FILE_EXTENSION_LEN;
```
Comment (michael-methner) [inline src/offlinelogstorage/dlt_offline_logstorage_behavior.c:382]: Please delete the two now unused defines from the header as well.

File: src/offlinelogstorage/dlt_offline_logstorage_behavior.c [hunk 3]
```diff
@@ -371,20 +370,15 @@ int dlt_logstorage_storage_dir_info(DltLogStorageUserConfig *file_config,
...
-        strncpy(suffix, ".dlt", suffix_len);
-    }
-
+    const char* suffix;
     for (i = 0; i < cnt; i++) {
-        const char* suffix = ".dlt";
+        if (config->gzip_compression) {
+            suffix = strdup(".dlt");
```
Comment (michael-methner) [inline src/offlinelogstorage/dlt_offline_logstorage_behavior.c:376]: strings allocated by strdup needs to be freed as well.

File: doc/design/dlt_offline_logstorage_design.md [hunk 4]
```diff
@@ -1,740 +0,0 @@
-# DLT Offline Logstorage Design Specification
```
[the remark this answers is not in the document]
Comment (minminlittleshrimp) [reply doc/design/dlt_offline_logstorage_design.md:1]: Hello @michael-methner 
As we did in the past, non of the design docs should be upstreamed. This was committed by mistake, hence I remove.
You can check this dir: https://gitlab.adit-cloud.com/oss/dlt-daemon/-/tree/adit_v2.18.x_master/doc/design?ref_type=heads
Some internal proprietaries (CRQ s.a. neg filter) will not be upstreammed then, unless you have other suggestion.
Regards

File: src/offlinelogstorage/dlt_offline_logstorage_behavior.c (hunk 2 above)
Comment (minminlittleshrimp) [reply src/offlinelogstorage/dlt_offline_logstorage_behavior.c:382]: Removed DLT_OFFLINE_LOGSTORAGE_FILE_EXTENSION_LEN

**Comment by minminlittleshrimp** [comment]
Hello @michael-methner , @lvklevankhanh 
Please kindly review for me.
Thank you.

**Review by lvklevankhanh — APPROVED** [review]

File: src/offlinelogstorage/dlt_offline_logstorage_behavior.c [hunk 5]
```diff
@@ -371,20 +370,15 @@ int dlt_logstorage_storage_dir_info(DltLogStorageUserConfig *file_config,
...
-    if (config->gzip_compression) {
-        suffix_len = DLT_OFFLINE_LOGSTORAGE_GZ_FILE_EXTENSION_LEN;
-        strncpy(suffix, ".dlt.gz", suffix_len);
-    }
-    else {
-        suffix_len = DLT_OFFLINE_LOGSTORAGE_FILE_EXTENSION_LEN;
-        strncpy(suffix, ".dlt", suffix_len);
-    }
-
+    char* suffix;
```
Comment (lvklevankhanh) [inline src/offlinelogstorage/dlt_offline_logstorage_behavior.c:373]: ‘suffix’ should be initialized.

Comment (minminlittleshrimp) [reply src/offlinelogstorage/dlt_offline_logstorage_behavior.c]: Set suffix to NULL
